src/redux/slices/toastsSlice.ts

```diff
@@ -0,0 +1,52 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { logoutAction } from '../commonActions';
+
+export interface CustomToastType {
+  toastId: string;
+  message: string;
+  duration?: number;
+  type: string;
+}
+
+export interface ToastsSliceState {
+  customToasts?: CustomToastType[] | undefined;
+}
+
+const initialState: ToastsSliceState = {};
+
+export const toastsSlice = createSlice({
+  name: 'toastsSlice',
+  initialState,
+  reducers: {
+    setCustomToasts: (
+      state: ToastsSliceState,
+      action: PayloadAction<CustomToastType>
+    ) => {
+      state.customToasts = [...(state.customToasts || []), action.payload].map(
+        (toast, index) => ({
+          ...toast,
+          type: 'custom',
+          toastId: toast.toastId || `custom-toast-${index}`
+        })
+      );
+    },
+    removeCustomToasts: (
+      state: ToastsSliceState,
+      action: PayloadAction<string>
+    ) => {
+      state.customToasts = (state.customToasts || []).filter(
```

Comment:
If we initialize the state with an empty array and make sure we don't nullify it, we can drop these checks. Also you can check for null with state?.customToasts?.filter